File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mp;
        for (int i=0; i<nums.size(); i++){
            int temp = target - nums[i];
            // checks for the complement
            if (mp.find(temp) != mp.end()){
                // mp[temp] is the index of the complement element int he original array
                // temp is the complement element
                return {mp[temp], i};
            }
            // key:element, value:index
            mp.insert({nums[i],i});
        }
        return {};
        }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        storage_map = {}
        for i in range(0, len(nums)):
            if nums[i] in storage_map:
                # Element present
                return [i, storage_map[nums[i]]]
            complement = target - nums[i] 
            # We need to store the index with the complement key
            storage_map[complement] = i